### ISSUE TITLE
Add logging to the assembler

### ISSUE THREAD

**Issue by bobbinth** [0xPolygonMiden/miden-vm#1106]
It may be good to add some basic logging functionality to the assembler. We could use [log](https://crates.io/crates/log) crate for this similar to how we do it in `miden-vm` crate. When logging is enabled, we'd be able to see extra output from the assembler and we'd also be able to report warnings (for something like #515).

We can start out with some minimal logging at first - e.g., log when a new module is being compiled (and maybe a coupe of other things) - and then we can expand this later on.

**Comment by Overcastan** [comment]
@bobbinth Do we want to allow user to decide what level of logging to use? 

Currently in `prove` CLI command implementation we [use](https://github.com/0xPolygonMiden/miden-vm/blob/c097af7ca53da2026cec10d9245a006b42186643/miden/src/cli/prove.rs#L71) hardcoded `debug` level, so usage of the `debug!("...")`, `warn!("...")` and `error!("...")` macro instructions is equivalent to the usage of the standard `println!()`.

We can remove this restriction, so logging level [can be specified](https://docs.rs/env_logger/0.7.1/env_logger/#enabling-logging) with `RUST_LOG` environment variable on the VM execution stage:
```
$ RUST_LOG=debug target/optimized/miden run -i example.inputs -a example.masm
```
Disadvantage of this approach is that by default (if we don't specify the `RUST_LOG` variable) logging level is set to `error!`, so only error messages will be displayed. 
I can extend the CLI to be able to specify the logging level environment variable with new feature (something like `--log`), in that case we can provide the user with the ability to specify the logging level, but at the same time the default level can be set to  "debug".

**Comment by bobbinth** [comment]
A couple thoughts:

- Allowing the user to specify logging level is a good idea. I think it is OK to do it via an environment variable - but I wonder we should use a more custom name (e.g., `MIDEN_LOG`). Scanning through the docs of `env_logger`, seems like it should be possible.
- We should probably enable logging for all commands. Maybe move logger initialization [here](https://github.com/0xPolygonMiden/miden-vm/blob/main/miden/src/main.rs#L55)?.
- I think we should set the default log level to `warn`. If users want to see more, they could enable lower levels manually.

**Comment by Overcastan** [comment]
Closed by #1139.
